extensions/google-drive/src/db.ts

```diff
@@ -169,27 +171,79 @@ export const insertFile = (
   const insertStatement = `
       INSERT
         INTO files (name, path, displayPath, fileSizeFormatted, createdAt, updatedAt)
-        VALUES ("${name}", "${path}", "${displayPath}", "${fileSizeFormatted}", "${createdAt.toISOString()}", "${updatedAt.toISOString()}")
+        VALUES ("${name}", "${path}", "${displayPath}", "${fileSizeFormatted}", "${createdAt}", "${updatedAt}")
         ON CONFLICT (path) DO
           UPDATE SET name = EXCLUDED.name, displayPath = EXCLUDED.displayPath, fileSizeFormatted = EXCLUDED.fileSizeFormatted, createdAt = EXCLUDED.createdAt, updatedAt = EXCLUDED.updatedAt;`;
 
   db.run(insertStatement);
 };
 
-export const walkRecursivelyAndSaveFiles = (path: PathLike, db: Database): void => {
-  if (getExcludePaths().includes(path.toLocaleString())) return;
-  saveFilesInDirectory(path, db);
-  getDirectories(path).map((dir) => walkRecursivelyAndSaveFiles(dir, db));
+const listFilesAndInsertIntoDb = async (path: string, db: Database, toast: Toast): Promise<void> => {
+  const updateToastMessage = throttledUpdateToastMessage(toast);
+
+  let totalFiles = 0;
+
+  if (!pathExists(path)) {
+    // TODO: message that path does not exist
+    return;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  for await (const _ of fg.stream([join(path, "**")], {
+    ignore: ["**/.Trashes", "**/.TemporaryItems", "**/node_modules", "**/.git", "**/tmp"],
+    dot: true,
+    suppressErrors: true,
+  })) {
+    totalFiles += 1;
+    updateToastMessage(`Counting files: ${totalFiles}`);
+  }
+
+  let filesIndexed = 0;
+
+  for await (const file of fg.stream([join(path, "**")], {
+    ignore: ["**/.Trashes", "**/.TemporaryItems", "**/node_modules", "**/.git", "**/tmp"],
+    dot: true,
+    suppressErrors: true,
+    stats: true,
+  })) {
+    const { name, path, stats } = file as unknown as Entry;
```

Comment:
The `file` has `string | Buffer` type here. Can we explicitly switch-case over these ADTs to avoid any runtime errors?